map/bookmark_helpers.cpp

```diff
@@ -370,33 +371,32 @@ std::string GetKMLorGPXPath(std::string const & filePath)
     if (kmlData == nullptr)
       return {};
 
-    fileSavePath = GenerateValidAndUniqueFilePathForKML(base::FileNameFromFullPath(filePath));
+    std::string fileSavePath = GenerateValidAndUniqueFilePathForKML(base::FileNameFromFullPath(filePath));
     if (!SaveKmlFileByExt(*kmlData, fileSavePath))
       return {};
+    return {fileSavePath};
   }
   else if (fileExt == kKmzExtension)
   {
-    // Extract KML file from KMZ archive and save to temp KML with unique name.
+    // Extract KML files from KMZ archive and save to temp KMLs with unique name.
+    std::vector<std::string> kmlFilePaths;
     try
     {
       ZipFileReader::FileList files;
       ZipFileReader::FilesList(filePath, files);
-      std::string kmlFileName;
       std::string ext;
-      for (size_t i = 0; i < files.size(); ++i)
+      for (const auto& file : files)
       {
-        ext = GetLowercaseFileExt(files[i].first);
+        std::string kmlFileName = file.first;
```

Comment:
No need to create a copy of the string here.
```suggestion
        auto const & kmlFileName = file.first;
```

Reply:
Fixed